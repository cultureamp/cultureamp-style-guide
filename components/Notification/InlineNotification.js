// @flow
import * as React from 'react';
import GenericNotification from './GenericNotification';

type Props = {|
  type: 'affirmative' | 'informative' | 'warning' | 'negative',
  title: string,
  children: React.Node,
  persistent?: boolean,
  onHide?: () => void,
|};

const InlineNotification = (props: Props) => (
  <GenericNotification style="normal" {...props} />
);

export default InlineNotification;
