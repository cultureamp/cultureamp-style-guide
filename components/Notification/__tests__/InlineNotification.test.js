import React from 'react';
import { mount } from 'enzyme';
import InlineNotification from '../InlineNotification';

test('The basic notification renders correctly', () => {
  const notification = mount(
    <InlineNotification type="cautionary" title="Warning">
      Something has gone wrong
    </InlineNotification>
  );
  expect(notification.find('.cancel')).toHaveLength(1);
  expect(notification).toMatchSnapshot();
});

test('Persistent versions of the notifications work', () => {
  const notification = mount(
    <InlineNotification type="negative" title="Error" persistent>
      Something has gone wrong
    </InlineNotification>
  );
  expect(notification.find('.cancel')).toHaveLength(0);
  expect(notification).toMatchSnapshot();
});
