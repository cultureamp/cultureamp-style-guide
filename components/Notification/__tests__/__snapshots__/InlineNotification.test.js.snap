// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Persistent versions of the notifications work 1`] = `
<InlineNotification
  persistent={true}
  title="Error"
  type="negative"
>
  <GenericNotification
    autohide={false}
    persistent={true}
    style="inline"
    title="Error"
    type="negative"
  >
    <div
      className="notification negative inline hidden"
      onTransitionEnd={[Function]}
      style={
        Object {
          "marginTop": undefined,
        }
      }
    >
      <div
        className="icon"
      >
        <Icon
          icon={null}
          inheritSize={true}
          role="presentation"
        >
          <svg
            aria-hidden={true}
            className="icon inheritSize"
            focusable="false"
            role="presentation"
            viewBox="viewBox"
          >
            <use
              xlinkHref="#id"
            />
          </svg>
        </Icon>
      </div>
      <div
        className="textContainer"
      >
        <span
          className="title"
        >
          Error
        </span>
        <span
          className="textWrapper"
        >
          <span
            className="text"
          >
            Something has gone wrong
          </span>
        </span>
      </div>
    </div>
  </GenericNotification>
</InlineNotification>
`;

exports[`The basic notification renders correctly 1`] = `
<InlineNotification
  persistent={false}
  title="Warning"
  type="warning"
>
  <GenericNotification
    autohide={false}
    persistent={false}
    style="inline"
    title="Warning"
    type="warning"
  >
    <div
      className="notification warning inline hidden"
      onTransitionEnd={[Function]}
      style={
        Object {
          "marginTop": undefined,
        }
      }
    >
      <div
        className="icon"
      >
        <Icon
          icon={null}
          inheritSize={true}
          role="presentation"
        >
          <svg
            aria-hidden={true}
            className="icon inheritSize"
            focusable="false"
            role="presentation"
            viewBox="viewBox"
          >
            <use
              xlinkHref="#id"
            />
          </svg>
        </Icon>
      </div>
      <div
        className="textContainer"
      >
        <span
          className="title"
        >
          Warning
        </span>
        <span
          className="textWrapper"
        >
          <span
            className="text"
          >
            Something has gone wrong
          </span>
        </span>
      </div>
      <CancelButton
        onClick={[Function]}
      >
        <button
          className="cancel"
          onClick={[Function]}
        >
          <span
            className="cancelInner"
          >
            <Icon
              icon={null}
              inheritSize={true}
              role="img"
              title="close notification"
            >
              <svg
                aria-labelledby="id-2-title"
                className="icon inheritSize"
                focusable="false"
                role="img"
                viewBox="viewBox"
              >
                <title
                  id="id-2-title"
                >
                  close notification
                </title>
                <use
                  xlinkHref="#id"
                />
              </svg>
            </Icon>
          </span>
        </button>
      </CancelButton>
    </div>
  </GenericNotification>
</InlineNotification>
`;
